org.adempiere.pipo/src/org/adempiere/pipo2/PoFiller.java

```diff
@@ -229,7 +229,12 @@ public Object setTableReference(String qName) {
 						}
 					}
 				}
-				if (id != null && refTableName != null) {
+				if (id instanceof Number && ((Number)id).intValue() == 0) {
+					if (refTableName != null && MTable.isZeroIDTable(refTableName)) {
+						po.set_ValueNoCheck(columnName, id);
+						return id;
+					}
+				} else if (id != null && refTableName != null) {

```

Comment:
The check for null should come first, or it should be removed.

Reply:
This is fine since "id instanceof Number" will evaluate  to false if id is null

Reply:
but this one make NPE 
(Number)id).intValue()

Reply:
no, that's not the case here. Java do short circuit evaluation, if the first part is false, it wouldn't evaluate the second part of the logic expression.